x-pack/plugins/cases/server/client/cases/utils.test.ts

```diff
@@ -1372,47 +1368,103 @@ describe('utils', () => {
       },
     ];
 
-    it('returns empty array when no key is missing', () => {
+    it('adds missing custom fields correctly', () => {
       expect(
-        compareCustomFieldKeysAgainstConfiguration({
-          requestCustomFields,
-          configurationCustomFields,
+        fillMissingCustomFields({
+          customFields,
+          customFieldsConfiguration,
         })
-      ).toEqual([]);
+      ).toEqual([
+        customFields[0],
+        {
+          key: 'second_key',
+          type: CustomFieldTypes.TOGGLE,
+          value: null,
+        },
+      ]);
     });
 
-    it('returns an empty array if request has no custom fields', () => {
+    it('does not set to null custom fields that exists', () => {
       expect(
-        compareCustomFieldKeysAgainstConfiguration({
-          requestCustomFields: [],
-          configurationCustomFields,
+        fillMissingCustomFields({
+          customFields: [
+            customFields[0],
+            {
+              key: 'second_key',
+              type: CustomFieldTypes.TOGGLE,
+              value: true,
+            },
+          ],
+          customFieldsConfiguration,
         })
-      ).toEqual([]);
+      ).toEqual([
+        customFields[0],
+        {
+          key: 'second_key',
+          type: CustomFieldTypes.TOGGLE,
+          value: true,
+        },
+      ]);
     });
 
-    it('returns missing keys', () => {
+    it('returns all custom fields if they are more than the configuration', () => {
```

Comment:
At the point in which this function is called the validation is already done so this should never happen. Shouldn't we throw an error?

Reply:
Yes, but the tests are for `fillMissingCustomFields` which is tested in isolation. There is no connection without validation in these tests. We need to test that the `fillMissingCustomFields` is working as expected in all scenarios because it may be used in a different context in the future.